### ISSUE TITLE
Content of _parsers not clear / hard to follow

### ISSUE THREAD

**Issue by sklemer1** [IdentityPython/pyFF#147]
I would consider moving this line

https://github.com/IdentityPython/pyFF/blob/b5c9bfb3c68dfa9c14c6d7572e16c340242d84c9/src/pyff/samlmd.py#L155

to parse.py. It took me a full step-by-step debug run to finally understand why parsing SAML data doesn't end in

https://github.com/IdentityPython/pyFF/blob/b5c9bfb3c68dfa9c14c6d7572e16c340242d84c9/src/pyff/parse.py#L27

**Comment by leifj** [comment]
Sure - I'll take a PR !
